test/FunctionalTests/Microsoft.OData.Core.Tests/SelectedPropertiesNodeTests.cs

```diff
@@ -399,9 +399,9 @@ public void SpecificActionShouldBeSelected()
         }
 
         [Fact]
-        public void SpecificActionWithParametersShouldBeSelected()
+        public void UnqualifiedUnboundActionWithParametersShouldNotBeSelected()
         {
-            SelectedPropertiesNode.Create("Action(Edm.Int32)").Should().HaveAction(this.cityType, this.action);
+            SelectedPropertiesNode.Create("Action(Edm.Int32)", this.cityType, this.edmModel).Should().NotHaveAction(this.cityType, this.action);
         }

```

Comment:
the old version of this method created the selected properties node as the top-level node, which is valid for an unbound operation.  With this change, you are selecting an unbound action from a type, which is not valid.

Reply:
removed this invalid case.

---
In reply to: [226137870](https://github.com/OData/odata.net/pull/1286#discussion_r226137870) [](ancestors = 226137870)

Reply:
Actually, the original case was valid, because it could be creating selectedpropertiesnode for a top level action/action import. It was only your proposed change that nested bound it to this.cityType, that was invalid.

---
In reply to: [226493137](https://github.com/OData/odata.net/pull/1286#discussion_r226493137) [](ancestors = 226493137,226137870)